### PR TITLE
macs: add apple-virt, nixos-test system features

### PR DESCRIPTION
**Pull request by cole-h**

### CODE REVIEW

File: macs/nix-darwin.nix [hunk 1]
```diff
@@ -37,6 +37,12 @@ in
     "extra-experimental-features" = [ "nix-command" "flakes" ];
     max-jobs = 4;
     cores = 2;
+    "system-features" = [
+      "apple-virt"
+      "nixos-test"
+      "big-parallel"
+      "nixos-test"
```
Comment (vcunat) [inline macs/nix-darwin.nix]: One `"nixos-test"` should suffice 😄

Comment (cole-h) [reply macs/nix-darwin.nix]: I don't know what you're talking about, there's definitely only one.... (nice catch, thanks, derp)

**Comment by mweinelt** [comment]
I think this was tested, and you discussed it in the infra room, but I missed the outcome. Can someone bring me up to speed?

**Comment by cole-h** [comment]
It was applied to one machine (`cosmic-stud`) and it "worked" in that they weren't being cancelled / aborted / whatever'd anymore, but the tests taking advantage of this would just hang (c.f. https://github.com/NixOS/nixpkgs/pull/303597#issuecomment-2130121108 and https://github.com/NixOS/nixpkgs/pull/303597#issuecomment-2135345269)

**Comment by vcunat** [comment]
As it was always hanging, I rolled back the rhea's `machines` file, so that scheduler wouldn't try to run it.

**Comment by Mic92** [comment]
Is this good to merge?

**Comment by vcunat** [comment]
Running those jobs doesn't seem to work.  They hang and time out.

**Comment by cole-h** [comment]
Closing. Feel free to reopen if we want to spend time figuring out why they're hanging.

**Pull request closed without merging**